src/flamenco/runtime/program/fd_bpf_loader_v3_program.c

```diff
@@ -71,117 +94,234 @@ read_bpf_upgradeable_loader_state_for_program( fd_exec_txn_ctx_t * txn_ctx,
     .valloc  = txn_ctx->valloc,
   };
 
-  if ( fd_bpf_upgradeable_loader_state_decode( result, &ctx ) ) {
-    FD_LOG_DEBUG(("fd_bpf_upgradeable_loader_state_decode failed"));
+  if( fd_bpf_upgradeable_loader_state_decode( result, &ctx ) ) {
+    FD_LOG_DEBUG(( "fd_bpf_upgradeable_loader_state_decode failed" ));
     *opt_err = FD_EXECUTOR_INSTR_ERR_INVALID_ACC_DATA;
     return NULL;
   }
 
   return rec->const_meta;  /* UGLY!!!!! */
 }
 
-int write_bpf_upgradeable_loader_state( fd_exec_instr_ctx_t * instr_ctx, fd_pubkey_t const * program_acc, fd_bpf_upgradeable_loader_state_t const * loader_state) {
-  int err = 0;
-  ulong encoded_loader_state_size = fd_bpf_upgradeable_loader_state_size( loader_state );
+/* https://github.com/anza-xyz/agave/blob/574bae8fefc0ed256b55340b9d87b7689bcdf222/programs/bpf_loader/src/lib.rs#L105-L171 */
+int
+deploy_program( fd_exec_instr_ctx_t * instr_ctx,
+                uchar * const         programdata,
+                ulong                 programdata_size ) {   
+  FD_SCRATCH_SCOPE_BEGIN {
```

Comment:
You could also just have a single `FD_SCRATCH_SCOPE_[BEGIN,END]` in `fd_bpf_loader_v3_program_execute`, which might be cleaner.